### DISCUSSION TITLE
How to use with GoXLR?

### DISCUSSION THREAD

**Discussion by DoluTattoo** [wwmm/easyeffects#2700] [General]
### EasyEffects Version

7.1.0

### What package are you using?

Arch (easyeffects)

### Distribution

Manjaro Gnome 44.5

### Describe the bug

With my GoXLR I have 4 output channels which I would like to pin to different EasyEffects profiles.

Here is my GoXLR ouptus:
- Chat
- Music
- Game
- System

Here is my profiles
- Dolu Chat
- Dolu Music
- Dolu Game
- Dolu System

I included/exlcuded apps for each profile.
Then I did that:
![image](https://github.com/wwmm/easyeffects/assets/47056777/f705a180-1ddb-490c-9f8d-3536b6ca840e)

Thing is, everything is going through my default output: System

### Expected Behavior

Is there a way to get an output for each channels?

### Debug Log

No log

**Comment by violetmage** [comment]
easyeffects manages only one effects chain at a time. Profiles are just predefined effects chains that are automatically applied depending on which device easyeffects is sending output to / receiving input from. If you want, you can manually assign the output easyeffects uses in its settings.

It seems like you are trying to route different applications to different outputs. Usually, that means you will need to use a patchbay app like helvum

**Comment by wwmm** [comment]
@dolutattoo it is not clear to me what you want to do. Do you want different presets to be loaded depending on which GoXLR channel is being set as the system default device? This should probably be already working. But as @violetmage has said EasyEffects has only one effects pipeline. So you can't apply different effects for different apps at the same time. And unless you put a loopback device in the middle EasyEffects will play to only one of those devices.

**Comment by DoluTattoo** [comment]
Thanks for your answers @violetmage and @wwmm.

What I was trying to do is to have as many presets as channels of my GoXLR (so 4) so I can tweak the effect differently on any of the 4 outputs.

It's hard to explain (i'm french and i'm very new into this) but for example:

- Spotify send music to Easy Effects in `Profile 1` => then it send it to my `GoXLR Music` output
- Discord send voice sounds to Easy Effects in `Profile 2` => then it send it to my `GoXLR Chat` output
- GTA V send sound to Easy Effects in `Profile 3` => then it send it to my `GoXLR Game` output
- Chrome send sound to Easy Effects in `Profile 4` => then it send it to my `GoXLR System` output

The goal is to be able to tweak any of my GoXLR output with this awesome app...

I understand now that Easy Effects has 1 pipeline.
Could you linkme/tell me more about this Patchbay thing?

**Comment by wwmm** [comment]
Maybe if you look into `PipeWire's filter chain` you will be able to apply different effects for each output at the same time. But you won't be able to use EasyEffects effects there. Only the filters supported by PipeWire's filter chain. And if I am not mistaken all its setup happens through the editing of configuration files. No graphical interface.

**Comment by violetmage** [comment]
@wwmm maybe move this thread to discussions?

**Comment by wwmm** [comment]
>maybe move this thread to discussions?

This should definitely be in the discussion page. But I have no idea about how to move it there.

**Comment by wwmm** [comment]
Oh... I think I've found it.
